dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+     */
+    @Override
+    public Mono<Void> refreshIndex(HttpHeaders headers, RefreshRequest refreshRequest) {
+
+        return sendRequest(refreshRequest, DotReactiveElasticsearchClient.RequestCreator.indexRefresh(), RefreshResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#updateMapping(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest)
+     */
+    @Override
+    public Mono<Void> updateMapping(HttpHeaders headers, PutMappingRequest putMappingRequest) {
+
+        return sendRequest(putMappingRequest, DotReactiveElasticsearchClient.RequestCreator.putMapping(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#flushIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.flush.FlushRequest)
+     */
+    @Override
+    public Mono<Void> flushIndex(HttpHeaders headers, FlushRequest flushRequest) {
+
+        return sendRequest(flushRequest, DotReactiveElasticsearchClient.RequestCreator.flushIndex(), FlushResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.ReactiveElasticsearchClientCallback)
+     */
+    @Override
+    public Mono<ClientResponse> execute(ReactiveElasticsearchClientCallback callback) {
+
+        return this.hostProvider.getActive(Verification.LAZY) //
+                .flatMap(callback::doWithClient) //
+                .onErrorResume(throwable -> {
+
+                    if (throwable instanceof ConnectException) {
+
+                        return hostProvider.getActive(Verification.ACTIVE) //
+                                .flatMap(callback::doWithClient);
+                    }
+
+                    return Mono.error(throwable);
+                });
+    }
+
+    @Override
+    public Mono<Status> status() {
+
+        return hostProvider.clusterInfo() //
+                .map(it -> new DotReactiveElasticsearchClient.ClientStatus(it.getNodes()));
+    }
+
+    // --> Private Response helpers
+
+    private static GetResult getResponseToGetResult(GetResponse response) {
+
+        return new GetResult(response.getIndex(), response.getType(), response.getId(), response.getSeqNo(),
+                response.getPrimaryTerm(), response.getVersion(), response.isExists(), response.getSourceAsBytesRef(),
+                response.getFields(), null);
+    }
+
+    // -->
+
+    private <Req extends ActionRequest, Resp> Flux<Resp> sendRequest(Req request, Function<Req, Request> converter,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'request' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)